networkx/algorithms/boundary.py

```diff
@@ -89,9 +89,13 @@ def edge_boundary(G, nbunch1, nbunch2=None, data=False, keys=False,
     # implemented by using the `not in` operator, instead of by creating
     # an additional set and using the `in` operator.
     if nbunch2 is None:
-        return (e for e in edges if e[1] not in nset1)
+        return (e for e in edges
```

Comment:
This could be `return (e for e in edges if (e[0] in nset1) ^ (e[1] in nset1))`, using the bitwise xor operator.